### PR TITLE
HDMI audio dynamic routing

### PR DESCRIPTION
**Pull request by gkdeepa**
To get the HDMI audio device and route
to the ON device

Tracked-On: OAM-98195
Signed-off-by: Deepa G K <g.k.deepa@intel.com>

### CODE REVIEW

File: hdmi/tinyaudio_hw.c [hunk 1]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+        ALOGE(" Failed to open mixer\n");
+        return -1;
+    }
+    for(i=0; i < MAX_HDMI_DEVICES; i++ ) {
+       char ctl_name[100] ;
+       enum mixer_ctl_type type;
+       unsigned int num_values;
+       sprintf(ctl_name, "HDMI/DP,pcm=%d Jack", i);
+       ctl = mixer_get_ctl_by_name(mixer, ctl_name);
+       if(ctl) {
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: introduce a space between if and (

File: hdmi/tinyaudio_hw.c [hunk 2]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+           type = mixer_ctl_get_type(ctl);
+           num_values = mixer_ctl_get_num_values(ctl);  
+           for (j = 0; j < num_values; j++) {
+               switch (type) {
+                   case MIXER_CTL_TYPE_BOOL:
+                   ALOGV("mixer_ctl_get_value %s", mixer_ctl_get_value(ctl, j) ? "On" : "Off");
+                   if(mixer_ctl_get_value(ctl, j)) {
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: introduce a space between if and (

File: hdmi/tinyaudio_hw.c [hunk 3]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+                       ALOGV(" status of device %d is ON",i);
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: introduce a space between , and i

File: hdmi/tinyaudio_hw.c [hunk 4]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+                       ALOGE("%s exit",__func__);
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: introduce a space between , and __func__

File: hdmi/tinyaudio_hw.c [hunk 5]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+    bool device_status;
 
+    ALOGV("%s enter",__func__);
+    card = get_card_number_by_name("PCH"); 
+    mixer = mixer_open(card);
+    if (!mixer) {
+        ALOGE(" Failed to open mixer\n");
+        return -1;
+    }
+    for(i=0; i < MAX_HDMI_DEVICES; i++ ) {
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: - introduce a space between for and (
- Remove space between i++ and )

File: hdmi/tinyaudio_hw.c [hunk 6]
```diff
@@ -228,25 +232,23 @@ static int start_output_stream(struct stream_out *out)
...
-        adev->card = DEFAULT_CARD;
+        adev->card = DEFAULT_CARD; 
+        adev->device = parse_hdmi_device_number();
+        /*Keeping the condtion check only for EHL, as it is not verified*/   
         property_get("ro.vendor.hdmi.audio", value, "0");
         if (!strcmp(value,"ehl")) {
             adev->device = DEFAULT_DEVICE_EHL;
-        } else {
-            adev->device = DEFAULT_DEVICE;
+            ALOGV("%s : Setting default card/ device %d,%d",__func__,adev->card,adev->device);
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: introduce a space between , and __func__ . Same comment applies for , and adev->card etc

File: hdmi/tinyaudio_hw.c [hunk 7]
```diff
@@ -228,25 +232,23 @@ static int start_output_stream(struct stream_out *out)
...
     struct pcm_params *params;
     int device = 0;
-
+    
     ALOGV("%s enter",__func__);
 
     if ((adev->card < 0) || (adev->device < 0)){
         char value[PROPERTY_VALUE_MAX];
-
+
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c:238]: seems a whitespace got introduced here. please check

File: hdmi/tinyaudio_hw.c [hunk 8]
```diff
@@ -228,25 +232,23 @@ static int start_output_stream(struct stream_out *out)
     struct audio_device *adev = out->dev;
     struct pcm_params *params;
     int device = 0;
-
+
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: seems whitespaces got introduced here. please check

File: hdmi/tinyaudio_hw.c [hunk 9]
```diff
@@ -228,25 +232,23 @@ static int start_output_stream(struct stream_out *out)
...
 
     if ((adev->card < 0) || (adev->device < 0)){
         char value[PROPERTY_VALUE_MAX];
-
+ 
         /*this will be updated once the hot plug intent
           sends these information.*/
-        adev->card = DEFAULT_CARD;
+        adev->card = DEFAULT_CARD; 
+        adev->device = parse_hdmi_device_number();
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c:242]: Please take care of error  case(parse_hdmi_device_number returning -1)

File: hdmi/tinyaudio_hw.c [hunk 10]
```diff
@@ -228,25 +232,23 @@ static int start_output_stream(struct stream_out *out)
...
+        /*Keeping the condtion check only for EHL, as it is not verified*/
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c]: condtion => condition

File: hdmi/tinyaudio_hw.c [hunk 11]
```diff
@@ -419,7 +421,51 @@ static int out_set_parameters(struct audio_stream *stream, const char *kvpairs)
...
+               switch (type) {
+                   case MIXER_CTL_TYPE_BOOL:
+                   ALOGV("mixer_ctl_get_value %s", mixer_ctl_get_value(ctl, j) ? "On" : "Off");
+                   if(mixer_ctl_get_value(ctl, j)) {
+                       ALOGV(" status of device %d is ON",i);
+                       ALOGE("%s exit",__func__);
+                       return i;
+                   }
+                   break;
+                   default:
```
Comment (JeevakaPrabu) [inline hdmi/tinyaudio_hw.c:463]: blank line between 459 and 460 line would be good

**Review by JeevakaPrabu — CHANGES_REQUESTED** [review]

**Review by JeevakaPrabu — APPROVED** [review]

**Review by padmashree9110 — APPROVED** [review]

**Pull request merged**